sdk/keyvault/azure-security-keyvault-perf/src/main/java/com/azure/security/keyvault/secrets/perf/ListSecretsTest.java

```diff
@@ -10,16 +10,43 @@
 import java.util.UUID;
 
 public class ListSecretsTest extends SecretsTest<PerfStressOptions> {
+    private String[] _secretNames;
+
     public ListSecretsTest(PerfStressOptions options) {
         super(options);
     }
 
+    @Override
     public Mono<Void> globalSetupAsync() {
-        return super.globalSetupAsync()
-            .then(Flux.range(0, options.getCount())
-                .map(i -> "listSecretsPerfTest-" + UUID.randomUUID())
-                .flatMap(b -> secretAsyncClient.setSecret(b, b))
-                .then());
+        super.globalSetupAsync().block();
+
+        // Validate that vault contains 0 secrets (including soft-deleted secrets), since additional secrets
+        // (including soft-deleted) impact performance.
+        if (secretClient.listPropertiesOfSecrets().iterator().hasNext() ||
+            secretClient.listDeletedSecrets().iterator().hasNext()) {
+
+            throw new RuntimeException("KeyVault " + secretClient.getVaultUrl() + "must contain 0 " +
```

Comment:
Should we consider deleting all secrets and then continue instead of throwing?

Reply:
Return `Mono.error()` instead.

>Should we consider deleting all secrets and then continue instead of throwing?

@vcolin7 We should not automatically delete just to be safe that we are not deleting something that shouldn't be. The executor of perf tests should ensure that there are no secrets before running the tests. Alternatively, we could add a flag to the test to delete all secrets before starting the test.

Reply:
@srnagar covered most of it.  It's dangerous for tests to delete resources they didn't create, so we just check and throw instead.  Perf tests should clean up after themselves, so you should only hit this if an earlier perf test crashed (or somehow failed to cleanup), or if you ran against the wrong KeyVault (and we **really** don't want to delete everything then).